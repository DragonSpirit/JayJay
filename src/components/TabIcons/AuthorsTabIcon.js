import React from 'react'
import PropTypes from 'prop-types'
import { Icon } from 'react-native-elements'

export const AuthorsTabIcon = ({ focused }) => (<Icon name='tag-faces' color={focused ? '#184fff' : '#000'} />)

AuthorsTabIcon.propTypes = {
  focused: PropTypes.bool,
}

AuthorsTabIcon.defaultProps = {
  focused: false,
}
