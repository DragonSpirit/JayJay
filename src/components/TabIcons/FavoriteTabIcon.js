import React from 'react'
import PropTypes from 'prop-types'
import { Icon } from 'react-native-elements'

export const FavoriteTabIcon = ({ focused }) => (<Icon name='star' color={focused ? '#184fff' : '#000'} />)

FavoriteTabIcon.propTypes = {
  focused: PropTypes.bool,
}
