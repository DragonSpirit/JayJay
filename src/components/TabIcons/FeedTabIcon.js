import React from 'react'
import PropTypes from 'prop-types'
import { Icon } from 'react-native-elements'

export const FeedTabIcon = ({ focused }) => (<Icon name='format-align-justify' color={focused ? '#184fff' : '#000'} />)

FeedTabIcon.propTypes = {
  focused: PropTypes.bool,
}
