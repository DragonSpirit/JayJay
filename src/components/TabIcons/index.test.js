import React from 'react'
import renderer from 'react-test-renderer'

import { AuthorsTabIcon } from './AuthorsTabIcon'
import { FavoriteTabIcon } from './FavoriteTabIcon'
import { FeedTabIcon } from './FeedTabIcon'

describe('TabIcons', () => {
  it('AuthorsTabIcon renders correctly', () => {
    const tree = renderer.create(<AuthorsTabIcon />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('FavoriteTabIcon renders correctly', () => {
    const tree = renderer.create(<FavoriteTabIcon />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('FeedTabIcon renders correctly', () => {
    const tree = renderer.create(<FeedTabIcon />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
