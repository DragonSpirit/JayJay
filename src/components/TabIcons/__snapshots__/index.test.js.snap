// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthorsTabIcon renders correctly 1`] = `
<View
  style={
    Array [
      false,
      false,
      false,
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
  underlayColor="white"
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#000",
          "fontSize": 24,
        },
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ],
        Object {
          "fontFamily": "Material Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    
  </Text>
</View>
`;

exports[`FavoriteTabIcon renders correctly 1`] = `
<View
  style={
    Array [
      false,
      false,
      false,
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
  underlayColor="white"
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#000",
          "fontSize": 24,
        },
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ],
        Object {
          "fontFamily": "FontAwesome",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    
  </Text>
</View>
`;

exports[`FeedTabIcon renders correctly 1`] = `
<View
  style={
    Array [
      false,
      false,
      false,
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
  underlayColor="white"
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#000",
          "fontSize": 24,
        },
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ],
        Object {
          "fontFamily": "Material Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    
  </Text>
</View>
`;
